iBridges.py

```diff
@@ -135,36 +137,34 @@ def login_function(self):
 
         """
         irods_env_file = self.irods_path.joinpath(self.envbox.currentText())
-        # Set the user-given filename for both the configuration and the Session
         self.context.irods_env_file = irods_env_file
-        self.context.irods_connector.irods_env_file = irods_env_file
         logging.debug('IRODS ENVIRONMENT FILE SET: %s', irods_env_file.name)
         self.envError.setText('')
-        if not (self.context.irods_environment.config and self.context.ienv_is_complete()):
+        if not self.context.ienv_is_complete():
             message = 'iRODS environment missing or incomplete.'
             logging.error(message)
             self.envError.setText(message)
             self.context.irods_environment.reset()
             self.passError.clear()
             self.setCursor(PyQt6.QtGui.QCursor(PyQt6.QtCore.Qt.CursorShape.ArrowCursor))
             return
-        irods_host = self.context.irods_environment.config.get('irods_host', '')
-        irods_port = self.context.irods_environment.config.get('irods_port', '')
+        irods_host = self.ienv.get('irods_host', '')
+        irods_port = self.ienv.get('irods_port', '')
         if not utils.utils.can_connect(irods_host, irods_port):
             message = 'No network connection to server: %s:%s'
             logging.warning(message, irods_host, irods_port)
             self.envError.setText(message % (irods_host, irods_port))
             self.setCursor(PyQt6.QtGui.QCursor(PyQt6.QtCore.Qt.CursorShape.ArrowCursor))
             return
         self.setCursor(PyQt6.QtGui.QCursor(PyQt6.QtCore.Qt.CursorShape.WaitCursor))
-        self.context.irods_connector.use_icommands = self.use_icommands
-        self.context.ibridges_configuration.config['last_ienv'] = irods_env_file.name
+        self.conn.use_icommands = self.use_icommands
+        self.conf['last_ienv'] = irods_env_file.name
```

Comment:
Here we actively change a local instance of the config, without changing it for the rest of the program. 
I think the single source of truth (singleton) should be updated here.

Reply:
You are very correct!  A property would prevent such an oversight.  I will revert that line.

Reply:
As a note, _if_ the `Context.ibridges_configuration` (or the `Context.irods_environment` for that matter) is reinitialized with a new `JSONConfig` at some point, this will break the connection as it will be a new dictionary instance.  This is _not_ done and there is no reason to do it (`JSONConfig`s are `reset()` not replaced between sessions), but a programmer unaware of this _feature_ could cause issues in this way.